### PR TITLE
Use `DateTimeOffset` everywhere instead of `string`

### LINKED ISSUE #524: [BUG]: Inconsistent types on WebHook models
### What happened?

Different types exposes similar properties using different underlying types, I was looking at the `*_at` properties in the class
[WorkflowJob](https://github.com/octokit/webhooks.net/blob/b3bb0e872ad8e09f9b11c2f2e29ad0022244494e/src/Octokit.Webhooks/Models/WorkflowJobEvent/WorkflowJob.cs#L62-L75) and seems like those properties are exposed as string while most of the time they're exposes as `DateTimeOffset`.

Having a quick look [here](https://github.com/search?q=repo%3Aoctokit%2Fwebhooks.net+_at%22%29&type=code&p=3) I found some more occurrences [here](https://github.com/octokit/webhooks.net/blob/b3bb0e872ad8e09f9b11c2f2e29ad0022244494e/src/Octokit.Webhooks/Models/RepositoryRulesetEvent/RepositoryRuleset.cs#L43-L47) and [here](https://github.com/octokit/webhooks.net/blob/b3bb0e872ad8e09f9b11c2f2e29ad0022244494e/src/Octokit.Webhooks/Models/WorkflowJobEvent/WorkflowJobStep.cs#L20-L21)

Can we unify those and use `DateTimeOffset` everywhere? This will be a breaking change so it requires a major version bump.

### Versions

2.2.2

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by JamieMagee**
Resolves #524 

----

### Before the change?

* An inconsistent mix of `string` and `DateTimeOffset` for `*_at` fields.

### After the change?

* Use `DateTimeOffset` everywhere instead of `string`

### Pull request checklist
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)

### Does this introduce a breaking change?

Please see our docs on [breaking changes](https://github.com/octokit/.github/blob/master/community/breaking_changes.md) to help!

- [x] Yes
- [ ] No

----

### CODE REVIEW

**Review by kfcampbell — APPROVED** [review]
@JamieMagee I think this is a good step, thank you! I'll leave it to your discretion when to merge/how to release this.

**Comment by ilmax** [comment]
Hey @JamieMagee, anything blocking this PR?

**Comment by JamieMagee** [comment]
@ilmax I cut this as v3.0.0

**Pull request merged**
